src/main/java/com/puppycrawl/tools/checkstyle/utils/TokenUtil.java

```diff
@@ -305,6 +325,18 @@ public static boolean isOfType(DetailAST ast, int... types) {
         return ast != null && isOfType(ast.getType(), types);
     }
 
+    /**
+     * Determines if the AST belongs to the given types.
+     *
+     * @param ast the AST node to check
+     * @param types the acceptable types
+     *
+     * @return true if type matches one of the given types.
+     */
+    public static boolean isOfType(DetailAST ast, List<Integer> types) {
+        return ast != null && isOfType(ast.getType(), types);
+    }
```

Comment:
> ast != null

This condition will not get covered by real check execution so pitest will complain. The real usage of this method in the checks always sends a `ast` that is always not null so this mutation will survive.

@romani  Do you agree with having a pure JUnit for this case like in: 

https://github.com/checkstyle/checkstyle/blob/15de92ae48ddea70c478d948bf2258906eb34b74/src/test/java/com/puppycrawl/tools/checkstyle/utils/TokenUtilTest.java#L362-L383


This condition is just a safety from NPE but there is no real execution case where we send a null to this method. What do you think?

Reply:
No pure unit testing.
All through Input files.

Reply:
I want to confirm my understanding of the issue. The problem with the changes I made is that the `ast != null` condition is not covered during either JUnit tests or normal execution. I considered adding something like the following in the `public void testIsOfTypeFalse()` method:

```java
final DetailAstImpl astForTest3 = null; 
List<Integer> typesList = List.of(TokenTypes.LITERAL_FOR, TokenTypes.LITERAL_IF, TokenTypes.LITERAL_ELSE);
final boolean result4 = TokenUtil.isOfType(astForTest3, typesList);
assertWithMessage("Token type should not match") 
    .that(result4) 
    .isFalse(); 
```
However, since you mentioned avoiding pure unit testing, should I think of a different approach to ensure the `ast != null` condition is covered?

Reply:
Then there will not be a nullity check in this method. However,  This will always be checked by the caller (We are sure that the `ast` passed in this method will not be null). So IMO,  yes we can only have `public static boolean isOfType(int type, List<Integer> types)`.

Reply:
We are not a Library for other on universal methods. Better to keep code that we know why it exists. It in future some Check would benefit from null checking, we will add such method and it will naturally cover such edge cases.

Reply:
We can modify the return value of `isSwitchExpression` to: 
```java
return !TokenUtil.isOfType(ast.getParent().getType(), switchStatementParents);
```
If we are certain that the parent of `ast` is always not null, otherwise, I don't currently see a way to avoid the null check

Reply:
If ast is not null, let's not make check for it.
Code coverage will demand coverage for all.